main.tf

```diff
@@ -1,24 +1,50 @@
 locals {
-  name          = "ibm-db2"
+  name          = "ibm-cpd-db2-instance"
+  subscription_name  = "ibm-cpd-db2-subscription"
   bin_dir       = module.setup_clis.bin_dir
-  yaml_dir      = "${path.cwd}/.tmp/${local.name}/chart/${local.name}"
+  //yaml_dir      = "${path.cwd}/.tmp/${local.name}/chart/${local.name}"
+  subscription_yaml_dir = "${path.cwd}/.tmp/${local.name}/chart/${local.subscription_name}"
+  instance_yaml_dir = "${path.cwd}/.tmp/${local.name}/chart/${local.name}"
   service_url   = "http://${local.name}.${var.namespace}"
   sa_name       = "ibm-db2-ibm-db2"
   layer = "services"
+  operator_type  = "operators"
   type  = "instances"
   application_branch = "main"
   namespace = var.namespace
   layer_config = var.gitops_config[local.layer]
-  values_content = {
-  }
+  subscription_content = {
+    name = "ibm-db2oltp-cp4d-operator-catalog-subscription"
+    operator_namespace = var.operator_namespace
+    syncWave = "-5"
+    spec = {
+      channel = var.channel
+      installPlanApproval = "Automatic"
+      name = "ibm-db2oltp-cp4d-operator"
+      source = "ibm-operator-catalog"
+      sourceNamespace = var.subscription_source_namespace        
+    }       
+  } 
   
- }
+  instance_content = {
+    name = "db2oltp-cr2"
+    cpd_namespace = var.cpd_namespace
+    spec = {
+      license = {
+        accept = "true"
+        license = var.license 
+        db_type = "db2oltp"
+        }      
+      }               
+    }   
+    //values_file = "values-${var.server_name}.yaml"
+}
 
 module setup_clis {
   source = "github.com/cloud-native-toolkit/terraform-util-clis.git"
 }
 
-module pull_secret {
+/*module pull_secret {
```

Comment:
If this is not needed, please remove the commented out code.

Reply:
environment issue: I understand, global auth has to be enabled using the entitlement key as per the cp4d document and talking to cp4d team, please advice?

Reply:
terraform-gitops-cp-db2-oltp has two charts ibm-cpd-db2-instance and ibm-cpd-db2-subscription, I referred to other modules, this is the way we are doing as sean has done in cp-bas

Reply:
removed commented code

Reply:
We should only add the pull secret to the namespace(s) where it is needed.  That way we don't have to reboot the entire cluster for it to take effect.   That's what I did for the cp4d-instance module.   that gets around the environment issue

Reply:
regarding the charts: if they are applied at different layers, we should keep as 2.  if they are applied at the exact same time, we should combine them.

Reply:
Cp4d team tried using pull secret but it doesn't work, that is what Cp4d team has settled it globally, we discussed it in office hours

Reply:
regarding the charts, they are applied at different layers, first subscription and then the instance, I am using syncWave for this